pkg/core/proxy/integrations/http/util.go

```diff
@@ -427,6 +427,37 @@ func isPassThrough(logger *zap.Logger, req *http.Request, destPort uint, opts mo
 				continue
 			}
 		}
+		if bypass.URLMethods != nil && len(bypass.URLMethods) != 0 {
+			for _, method := range bypass.URLMethods {
+				if method == req.Method {
+					passThrough = true
+					break
+				}
+			}
+			if !passThrough {
+				continue
+			}
+		}
+		if bypass.Headers != nil && len(bypass.Headers) != 0 {
+			for bypassHeaderKey, bypassHeaderValue := range bypass.Headers {
+				regex, err := regexp.Compile(bypassHeaderValue)
+				if err != nil {
```

Comment:
similarly for cases where multiple checks are to be enforced , how do we asssure that?